cmd/client-fs.go

```diff
@@ -765,33 +702,9 @@ func (f *fsClient) listInRoutine(contentCh chan<- *clientContent, isMetadata boo
 			if fi.Mode()&os.ModeSymlink == os.ModeSymlink {
 				fp := filepath.Join(fpath, fi.Name())
 				fi, e = os.Stat(fp)
-				if os.IsPermission(e) {
-					contentCh <- &clientContent{
-						Err: probe.NewError(PathInsufficientPermission{Path: pathURL.Path}),
-					}
-					continue
-				}
-				if os.IsNotExist(e) {
-					// Lstat makes no attempt to follow the broken link.
-					_, e = os.Lstat(fp)
-					contentCh <- &clientContent{
-						URL:  pathURL,
-						Size: -1,
-						Err:  probe.NewError(e),
-					}
-					continue
-				}
 				if e != nil {
-					if strings.Contains(e.Error(), "too many levels of symbolic links") {
-						contentCh <- &clientContent{
-							Err: probe.NewError(TooManyLevelsSymlink{
-								Path: pathURL.Path,
-							}),
-						}
-					} else {
-						contentCh <- &clientContent{
-							Err: probe.NewError(e),
-						}
+					contentCh <- &clientContent{
+						Err: f.toClientError(e, fp),
```

Comment:
Also, `toClientError` function handles only `PathInsufficientPermission` and `PathNotFound`. We should add cases for `TooManyLevelsSymlink`,`BrokenSymlink` etc.

Reply:
toClientError also handles TooManyLevelsSymlink - there is no error for BrokenSymlink @Praveenrajmani